app/src/androidTest/java/io/reark/rxgithubapp/advanced/data/stores/GitHubRepositoryStoreTest.java

```diff
@@ -198,6 +201,83 @@ public void getOnceAndStream_WithInitialValue_WithDelayedSubscription_ReturnsFir
         testSubscriber.assertReceivedOnNext(singletonList(value1));
     }
 
+    @Test
+    public void put_WithNewData_EmitsTrue() {
+        final GitHubRepository value = create(100, "repository1");
+
+        gitHubRepositoryStore.put(value)
+                .test()
+                .awaitTerminalEvent(Constants.Tests.PROVIDER_WAIT_TIME, TimeUnit.MILLISECONDS)
+                .assertValue(true);
+    }
+
+    @Test
+    public void put_WithDifferentData_OverExistingData_EmitsTrue() throws InterruptedException {
+        final GitHubRepository value1 = create(100, "repository1");
+        final GitHubRepository value2 = create(100, "repository2");
+        gitHubRepositoryStore.put(value1);
+        Thread.sleep(Constants.Tests.PROVIDER_WAIT_TIME);
+
+        gitHubRepositoryStore.put(value2)
+                .test()
+                .awaitTerminalEvent(Constants.Tests.PROVIDER_WAIT_TIME, TimeUnit.MILLISECONDS)
+                .assertValue(true);
+    }
+
+    @Test
+    public void put_WithIdenticalData_OverExistingData_EmitsFalse() throws InterruptedException {
+        final GitHubRepository value = create(100, "repository1");
+        gitHubRepositoryStore.put(value);
+        Thread.sleep(Constants.Tests.PROVIDER_WAIT_TIME);
+
+        gitHubRepositoryStore.put(value)
+                .test()
+                .awaitTerminalEvent(Constants.Tests.PROVIDER_WAIT_TIME, TimeUnit.MILLISECONDS)
+                .assertValue(false);
+    }
+
+    @Test
+    public void delete_WithNoData_EmitsFalse() {
+        gitHubRepositoryStore.delete(765)
+                .test()
+                .awaitTerminalEvent(Constants.Tests.PROVIDER_WAIT_TIME, TimeUnit.MILLISECONDS)
+                .assertCompleted()
+                .assertValue(false);
+    }
+
+    @Test
+    public void delete_WithData_DeletesData_AndEmitsTrue() throws InterruptedException {
+        final GitHubRepository value = create(100, "repository1");
+        gitHubRepositoryStore.put(value);
+        Thread.sleep(Constants.Tests.PROVIDER_WAIT_TIME);
+
+        AssertableSubscriber<Boolean> ts1 = gitHubRepositoryStore.delete(100).test();
+        Thread.sleep(Constants.Tests.PROVIDER_WAIT_TIME);
+        AssertableSubscriber<GitHubRepository> ts2 = gitHubRepositoryStore.getOnce(100).test();
+
+        ts1.assertCompleted()
+                .assertValue(true);
+        ts2.awaitTerminalEvent(Constants.Tests.PROVIDER_WAIT_TIME, TimeUnit.MILLISECONDS)
+                .assertCompleted()
+                .assertNoErrors()
+                .assertReceivedOnNext(singletonList(none()));
+    }
+
+    @Test
+    public void getOnceAndStream_ThenDelete_DoesNotEmit() throws InterruptedException {
```

Comment:
What's the purpose of this test? I am not sure what the stream might emit in case of delete.. though of course it does work as supposed.